ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java

```diff
@@ -120,21 +122,80 @@ private double getRemedialActionResultValue(RemedialActionResult remedialActionR
+        //mapping the history RAs to their IDs
+        Map<String, RemedialActionResult> raResultHistoryMap = new HashMap<>();
+        for (RemedialActionResult remedialActionResult : remedialActionResultHistoryList) {
+            raResultHistoryMap.put(remedialActionResult.getId(), remedialActionResult);
         }
-        Map<String, Double> firstMap = new HashMap<>();
-        for (RemedialActionResult remedialActionResult : firstList) {
-            firstMap.put(remedialActionResult.getId(), getRemedialActionResultValue(remedialActionResult));
+        //mapping the new RAs to their IDs
+        Map<String, RemedialActionResult> newRAResultMap = new HashMap<>();
+        for (RemedialActionResult remedialActionResult : newRemedialActionResultList) {
+            newRAResultMap.put(remedialActionResult.getId(), remedialActionResult);
+        }
+        //initializing an empty RA list which will contain the combination of both lists of RAs
+        List<RemedialActionResult> newRemedialActionResultHistoryList = new ArrayList<>();
+        /*
+        If an element is in one list but not the other (checking with the maps), we add it to the new list.
+        If it is in both lists (they will be set to different values) we combine them using the combineRAResults method.
+          If they end up canceling each other out (for instance the old RA could set a tap from 4 to 8 and the new RA
+          could set the same tap from 8 to 4), the combined RA will be empty and we do not add it to the new list.
+        */
+        for (RemedialActionResult remedialActionResult : newRemedialActionResultList) {
+            if (!raResultHistoryMap.containsKey(remedialActionResult.getId())) {
+                newRemedialActionResultHistoryList.add(remedialActionResult);
+            } else {
+                RemedialActionResult combinedRemedialActionResult = combineRAResults(raResultHistoryMap.get(remedialActionResult.getId()), remedialActionResult);
+                if (!combinedRemedialActionResult.getRemedialActionElementResults().isEmpty()) {
+                    newRemedialActionResultHistoryList.add(combinedRemedialActionResult);
+                }
+            }
         }
-        for (RemedialActionResult remedialActionResult : secondList) {
-            if (!firstMap.containsKey(remedialActionResult.getId()) ||
-                    Math.abs(firstMap.get(remedialActionResult.getId()) - getRemedialActionResultValue(remedialActionResult)) > 0.001) {
-                return false;
+        for (RemedialActionResult remedialActionResult : remedialActionResultHistoryList) {
+            if (!newRAResultMap.containsKey(remedialActionResult.getId())) {
+                newRemedialActionResultHistoryList.add(remedialActionResult);
             }
         }
-        return true;
+        remedialActionResultHistoryList = newRemedialActionResultHistoryList;
     }
```

Comment:
Can be hugely reduced by defining `remedialActionResultHistoryList` as a Map directly

Reply:
true